storage_service/locations/models/space.py

```diff
@@ -306,13 +317,22 @@ def _move_rsync(self, source, destination):
             raise StorageException(s)
 
     def _create_local_directory(self, path, mode=None):
-        """ Creates a local directory at 'path' with 'mode' (default 775). """
+        """
+        Creates directory structure for `path` with `mode` (default 775).
+        :param path: path to create the directories for.  Should end with a / or
+            a filename.
+        :param mode: (optional) Permissions to create the directories with
+            represented in octal (like bash or the stat module)
+        """
         if mode is None:
             mode = (stat.S_IRUSR + stat.S_IWUSR + stat.S_IXUSR +
                     stat.S_IRGRP + stat.S_IWGRP + stat.S_IXGRP +
                     stat.S_IROTH +                stat.S_IXOTH)
+        dir_path = os.path.dirname(path)
+        if not dir_path:
+            return
```

Comment:
I updated the docstring.  The original (intended?) usage is to pass the path to the file that needs directories, and this handles any directory creation required.  Eg. path/to/foo.txt creates path/to, but just foo.txt creates nothing because nothing needs creating.